redwood-lazylayout-compose/src/commonMain/kotlin/app/cash/redwood/lazylayout/compose/ReuseList.kt

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 2024 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.redwood.lazylayout.compose
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.movableContentOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+
+/**
+ * Execute [render] for each item in [items], while minimizing the amount of work the applier needs
+ * to do when elements are added and removed.
+ *
+ * This uses Compose's [movableContentOf] to match inserted items to removed items, and will reuse
+ * element subtrees where possible. This will also track moves of elements within the list, so long
+ * as the key function matches.
+ */
+@Composable
+public fun <T> ReuseList(
```

Comment:
This isn’t strictly a LazyLayout thing; could just as easily go in redwood-runtime. Or even in the individual guest applications. There’s nothing particularly Redwood-proprietary here.